compiler/src/dmd/cpreprocess.d

```diff
@@ -84,11 +85,17 @@ FileName preprocess(FileName csrcfile, ref const Loc loc, out bool ifile, OutBuf
             fatal();
         }
         //printf("C preprocess succeeded %s\n", ifilename.ptr);
-        ifile = true;
-        return FileName(ifilename);
+        auto readResult = File.read(ifilename);
+        File.remove(ifilename.ptr);
+        Mem.xfree(cast(void*)ifilename.ptr);
+        if (!readResult.success)
+            return DArray!ubyte();
+        return DArray!ubyte(readResult.extractSlice());
```

Comment:
What's the rationale for to not use `getFileContents` here?

Reply:
I am moving it to runPreprocessor() next so that it can be read instead from the stdout of the preprocessor run. That will resolve the temporary file problem. This PR is a step in that direction.